### PR TITLE
EN-9978: Allow running the domain report for a configurable period

### PR DESCRIPTION
**Pull request by housepage**

### CODE REVIEW

File: scriptlets/domain_report.js [hunk 1]
```diff
@@ -22,8 +24,8 @@ var millisecondsFromEpochToISODateString = function(milliseconds) {
 };
 
 var runAndWriteToFile = function () {
-    if (start == null || end == null) {
-        scriptlet.errors = "This scriptlet requires a start and end date";
+    if (start == null || end == null || period == null) {
+        scriptlet.errors = "This scriptlet requires a start data, an end date, and a period.";
```
Comment (raganw) [inline scriptlets/domain_report.js]: `s/data/date/`

File: scriptlets/domain_report.js [hunk 2]
```diff
@@ -7,8 +7,10 @@ var USED_TO_IMPLEMENT_SETS = "ANY VALUE WILL DO";
 var info = function () {
     var ret = {};
     ret['name'] = "Domain Report";
-    ret['description'] = "Monthly Report of all Site Metrics";
-    ret['params'] = { start: { class: "date", default: "2009-01-01"}, end: { class: "date", default: "2014-01-01"} };
+    ret['description'] = "Report of all Site Metrics";
+    ret['params'] = { start: { class: "date", default: "2009-01-01"},
```
Comment (raganw) [inline scriptlets/domain_report.js]: Why not take this opportunity to update the default dates to something more recent?

File: scriptlets/domain_report.js (hunk 1 above)
Comment (raganw) [inline scriptlets/domain_report.js]: Maybe provide allowable examples for `period`, also.

File: scriptlets/domain_report.js (hunk 2 above)
Comment (housepage) [inline scriptlets/domain_report.js]: Sure.

Comment (housepage) [inline scriptlets/domain_report.js]: Sure thing.

**Comment by raganw** [comment]
Looks good to me :shipit:

**Pull request merged**